tket2/src/rewrite_space/factory/kak.rs

```diff
@@ -0,0 +1,255 @@
+//! KAKResynthesis explorer for rewrite space exploration.
+use hugr::{
+    builder::{
+        endo_sig, DFGBuilder, Dataflow, DataflowHugr
+    }, 
+    extension::{prelude::qb_t, simple_op::MakeExtensionOp},
+    Hugr, HugrView, IncomingPort, OutgoingPort
+};
+use hugr::Direction;
+use tket2::{
+    portmatching::PatternMatch, rewrite_space::{
+        CommitFactory, IterMatchedWires, PersistentHugr, Walker
+    }, Tk2Op
+};
+use hugr::persistent::{PersistentWire, PatchNode};
+
+pub struct KAKResynthesis{
+    wires: Vec<PersistentWire>
+}
+
+impl IterMatchedWires for KAKResynthesis {
+    fn matched_wires(&self) -> impl Iterator<Item = &PersistentWire> + '_ {
+        self.wires[0].iter().chain(self.wires[1].iter())
+    }
+}
```

Comment:
There's something weird here: the way this is written worked for `wires: [Vec<PersistentWire>; 2]`, but it looks like you've changed it to a simple `Vec<_>`. In this case, `self.wires.iter()` should be all you need here.